Develop

RazieL-hub opened the pull request:
dell comments, add .env, dell unused imports

RazieL-hub committed:
fix views , celery, del comments

RazieL-hub committed:
fix views , celery, del comments

RazieL-hub committed:
fix views , celery, del comments

RazieL-hub committed:
fix views , celery, del comments

Niccolum commented on Support/authentication/views.py:
```diff
@@ -1,32 +1,53 @@
-                     f" "
-        send_activation_email.delay(serializer.validated_data.get('email'))
+                     f"http://{current_site}/authentication/verify-email/?token={str(token)}"
+        send_activation_email.delay(email_subject, email_body, os.getenv('EMAIL_HOST_USER'),
```
1. А нужно ли явно передавать EMAIL_HOST_USER? Мб подставлять его внутри?
2. Если всё таки нужно подставлять явно, то сохраняй его в настройках и получай из from django.conf import settings

RazieL-hub replied:
ИСПРАВЛЕНО

Niccolum commented on Support/authentication/views.py:
```diff
@@ -1,32 +1,53 @@
+    def get(self, request):
+        token = request.GET.get('token')
+        try:
+            payload = jwt.decode(token, os.getenv('SECRET_KEY'))
```
Secret_key лучше получать явно из настроек
Сейчас у тебя конфиг по факту размазан по всему коду. Если вдруг что поменяется (имя переменной среды, к примеру) или какая логика, с ним связанная, придётся искать по всему коду и думать исправить или нет.

RazieL-hub replied:
ИСПРАВЛЕНО

Niccolum commented on Support/authentication/views.py:
```diff
@@ -1,32 +1,53 @@
+        token = request.GET.get('token')
+        try:
+            payload = jwt.decode(token, os.getenv('SECRET_KEY'))
+            user = User.objects.get(id=payload['user_id'])
```
Нет обработки ошибки на несуществующего юзера

RazieL-hub replied:
Код пока вообще удалю

RazieL-hub replied:
он вообще пока не работает.

RazieL-hub replied:
Этот код хочу оставить для базы, потом допилить.

Niccolum commented on Support/taskmanager/views.py:
```diff
@@ -31,18 +32,19 @@ def post(self, request):
 class TicketListView(APIView):
     """Вывод полного списка тикетов"""
-    # permission_classes = [IsAuthenticated]
+    permission_classes = (AllowAny, )
```
Почему список ответов видят все?

RazieL-hub replied:
я с пермишинами балуюсь пока больше.

RazieL-hub committed:
fix setting.SECRET_KEY setting.EMAIL_HOST_USER

RazieL-hub committed:
correct views taskmanager on generic-class
fix name test1.json on fixture.json

RazieL-hub committed:
correct views taskmanager on generic-class
fix name test1.json on fixture.json
delete djoser, delete comments, delete notused imports

Niccolum commented on Support/taskmanager/views.py:
```diff
@@ -9,57 +10,37 @@
+    """Вывод конкретного тикетов"""
+
+    serializer_class = TicketDetailSerializer
+    queryset = Ticket.objects.filter()
```
Что за пустой filter?

RazieL-hub replied:
ну так, как явный pk  я не передаю, он берёт автоматом.

RazieL-hub replied:
ИСПРАВЛЕНО

RazieL-hub committed:
correct views taskmanager on generic-class
fix name test1.json on fixture.json
delete djoser, delete comments, delete notused imports
fix TicketDetailView all -->> filter()

RazieL-hub committed:
correct views taskmanager on generic-class
fix name test1.json on fixture.json
delete djoser, delete comments, delete notused imports
fix TicketDetailView all -->> filter()
